canvas_sdk/data/task.py

```diff
@@ -0,0 +1,51 @@
+from datetime import datetime
+from enum import Enum
+
+from canvas_sdk.data import Model
+from canvas_sdk.data.patient import Patient
+from canvas_sdk.data.staff import Staff
+from canvas_sdk.effects import Effect, EffectType
+
+
+class Task(Model):
+    class Status(Enum):
+        COMPLETED = "Completed"
+        CLOSED = "Closed"
+        OPEN = "Open"
+
+    id: str | None = None
+    assignee: Staff | None = None
+    patient: Patient | None = None
+    title: str | None = None
+    due: datetime | None = None
+    status: Status | None = None
+    comments: "list[TaskComment] | None" = None
+    labels: list[str] | None = None
+
+    def create(self) -> Effect:
+        return Effect(type=EffectType.CREATE_TASK, payload=self.model_dump_json_nested())
```

Comment:
```suggestion
        if self.id:
            raise ValueError("Cannot create a Task with an id")
        return Effect(type=EffectType.CREATE_TASK, payload=self.model_dump_json_nested())
```

Reply:
I added this to the `_validate_before_effect` method in the base `DataModel` class  that I just pushed up.